File: android/src/main/java/com/eko/RNBGDTaskConfig.java
package com.eko;

import java.io.Serializable;

public class RNBGDTaskConfig implements Serializable {
    public String id;
    public String url;
    public String destination;
    public String metadata = "{}";
    public String notificationTitle;
    public boolean reportedBegin;

    public RNBGDTaskConfig(String id, String url, String destination, String metadata, String notificationTitle) {
        this.id = id;
        this.url = url;
        this.destination = destination;
        this.metadata = metadata;
        this.notificationTitle = notificationTitle;
        this.reportedBegin = false;
    }
}

File: android/src/main/java/com/eko/RNBGDTaskConfig.kt
package com.eko

data class RNBGDTaskConfig(
  val id: String,
  val url: String,
  val destination: String,
  val metadata: String = "{}",
  val headers: Map<String, String> = emptyMap(),
  val expectedSha256: String? = null,
  var state: Int = DownloadConstants.TASK_RUNNING,
  var errorCode: Int = 0
)
